dbt_semantic_interfaces/objects/where_filter/filter_renderer.py

```diff
@@ -42,161 +47,86 @@ class FilterCallParameterSets:
-        """Parse the filter and extract the metric object call parameters.
-
-        An abbreviated example:
-
-        "{{ dimension('home_state_latest', entity_path=['user']) }} IN ('CA', 'HI', 'WA')"
-
-        ->
-
-        FilterCallParameterSets(
-            dimension_call_parameter_sets=(
-                DimensionCallParameterSet(
-                    entity_path=("user",),
-                    dimension_reference="home_state_latest",
-                ),
-            )
-            ...
-        )
-
-        """
+    def transform(self, where_filter: WhereFilter) -> FilterCallParameterSets:  # noqa: D
         dimension_call_parameter_sets: List[DimensionCallParameterSet] = []
         time_dimension_call_parameter_sets: List[TimeDimensionCallParameterSet] = []
         entity_call_parameter_sets: List[EntityCallParameterSet] = []
 
-        class _DummyCallRenderer(FilterFunctionCallRenderer):
-            def render_dimension_call(self, dimension_call_parameter_set: DimensionCallParameterSet) -> str:  # noqa: D
-                dimension_call_parameter_sets.append(dimension_call_parameter_set)
-                return FilterRenderer._DUMMY_PLACEHOLDER
-
-            def render_time_dimension_call(  # noqa: D
-                self, time_dimension_call_parameter_set: TimeDimensionCallParameterSet
-            ) -> str:
-                time_dimension_call_parameter_sets.append(time_dimension_call_parameter_set)
-                return FilterRenderer._DUMMY_PLACEHOLDER
-
-            def render_entity_call(self, entity_call_parameter_set: EntityCallParameterSet) -> str:  # noqa: D
-                entity_call_parameter_sets.append(entity_call_parameter_set)
-                return FilterRenderer._DUMMY_PLACEHOLDER
-
-        FilterRenderer.render(
-            templated_filter_sql=templated_filter,
-            call_renderer=_DummyCallRenderer(),
-        )
-
-        return FilterCallParameterSets(
-            dimension_call_parameter_sets=tuple(dimension_call_parameter_sets),
-            time_dimension_call_parameter_sets=tuple(time_dimension_call_parameter_sets),
-            entity_call_parameter_sets=tuple(entity_call_parameter_sets),
-        )
-
-    @staticmethod
-    def render(
-        templated_filter_sql: str,
-        call_renderer: FilterFunctionCallRenderer,
-    ) -> str:
-        """Render the templated SQL according to the supplied functions. See class docstring for an example.
-
-        Args:
-            templated_filter_sql: the templated SQL to render
-            call_renderer: Specifies how to render calls like dimension(...) in the template.
-
-        Returns:
-            The templated SQL rendered according to the supplied functions.
-        """
-
-        def _dimension_helper(dimension_name: str, entity_path: Sequence[str] = ()) -> str:
+        def _dimension_call(dimension_name: str, entity_path: Sequence[str] = ()) -> str:
```

Comment:
Oh these are returning str because of jinja, right?

Reply:
Yes.